language/resource-viewer/src/lib.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    cached_access_path_table::resource_vec_to_type_tag,
+    resolver::Resolver,
+    value::{MoveStruct, MoveType, MoveValue, StructDef},
+};
+use anyhow::{anyhow, Result};
+use libra_state_view::StateView;
+use libra_types::{
+    access_path::AccessPath,
+    account_address::AccountAddress,
+    account_state::AccountState,
+    contract_event::ContractEvent,
+    language_storage::StructTag,
+};
+use move_core_types::identifier::Identifier;
+use std::{
+    collections::btree_map::BTreeMap,
+    convert::TryFrom,
+    fmt::{Display, Formatter},
+};
+
+pub use cached_access_path_table::update_mapping;
+use move_vm_types::loaded_data::types::{FatStructType, FatType};
+
+mod cached_access_path_table;
+mod module_cache;
+mod resolver;
+pub mod value;
+
+#[derive(Debug)]
+pub struct AnnotatedAccountStateBlob(BTreeMap<StructTag, AnnotatedMoveResource>);
+
+#[derive(Debug)]
+pub struct AnnotatedMoveResource {
+    type_: StructTag,
+    value: BTreeMap<Identifier, AnnotatedMoveValue>,
+}
+
+#[derive(Debug)]
+pub enum AnnotatedMoveValue {
+    U8(u8),
+    U64(u64),
+    U128(u128),
+    Bool(bool),
+    Address(AccountAddress),
+    Vector(Vec<AnnotatedMoveValue>),
+    Bytes(Vec<u8>),
```

Comment:
My hope is that this `AnnotatedMoveValue` could be consumed by clients to make better sense of the on chain data, and this will have no resemblance of a canonical representation. The enum definition in `value.rs` will be the real canonical form that I'm planning to move to `move_core_types` in the future.

Reply:
Continuing on what we talked offline: could we add a comment here to indicate this is more like a genesis/resource viewer specific thing, and discourage new clients from using it unless they really know what they are doing? Maybe point them to the MoveValue?